sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -716,6 +716,35 @@ object PushFoldableIntoBranches extends Rule[LogicalPlan] with PredicateHelper {
         c.copy(
           branches.map(e => e.copy(_2 = b.withNewChildren(Array(left, e._2)))),
           Some(b.withNewChildren(Array(left, elseValue.getOrElse(Literal(null, c.dataType))))))
+
+      case s @ StaticInvoke(_, _, "contains" | "startsWith" | "endsWith",
```

Comment:
It seems we can define an `object SupportedBinaryExpr` and implement an `unapply` method to match supported binary expressions and `StaticInvoke` that represents string/binary predicates.

Note, we should not only check the function name, but also the object to call static functions.

Reply:
How about current?